src/commands/deploy.ts

```diff
@@ -119,8 +120,22 @@ export async function deploy(context: IActionContext, confirmDeployment: boolean
     if (confirmDeployment && siteConfig.scmType !== constants.ScmType.LocalGit && siteConfig !== constants.ScmType.GitHub) {
         const warning: string = `Are you sure you want to deploy to "${node.treeItem.client.fullName}"? This will overwrite any previous deployment and cannot be undone.`;
         context.properties.cancelStep = 'confirmDestructiveDeployment';
-        const deployButton: vscode.MessageItem = { title: 'Deploy' };
-        await ext.ui.showWarningMessage(warning, { modal: true }, deployButton, DialogResponses.cancel);
+        const items: vscode.MessageItem[] = [{ title: 'Deploy' }, DialogResponses.cancel];
+        if (defaultWebAppToDeploy) {
+            items.push({ title: 'Reset default' });
+        }
+        const result: vscode.MessageItem = await ext.ui.showWarningMessage(warning, { modal: true }, ...items);
+        if (result.title === 'Reset default') {
+            // tslint:disable-next-line:no-non-null-assertion
+            const localRootPath = currentWorkspace!.uri.fsPath;
+            const settingsPath = path.join(localRootPath, '.vscode', 'settings.json');
+            const doc = await vscode.workspace.openTextDocument(vscode.Uri.file(settingsPath));
+            vscode.window.showTextDocument(doc);
+            const workspaceConfiguration: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration(constants.extensionPrefix, vscode.Uri.file(fsPath));
+            await workspaceConfiguration.update(constants.configurationSettings.defaultWebAppToDeploy, '');
```

Comment:
Is it better to just delete line 134? I don't need it because of [this line](https://github.com/Microsoft/vscode-azureappservice/blob/da506f428c3e9b1dd4a8004912ea4bcb2ff74b5a/src/commands/deploy.ts#L111)

Reply:
Well that line should also be using this: https://github.com/Microsoft/vscode-azurefunctions/blob/master/src/ProjectSettings.ts#L74